api/apps/api/src/modules/clone/export/application/complete-piece.handler.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  CommandHandler,
+  EventPublisher,
+  IInferredCommandHandler,
+} from '@nestjs/cqrs';
+import { Logger } from '@nestjs/common';
+
+import { CompletePiece } from './complete-piece.command';
+import { ExportRepository } from './export-repository.port';
+
+@CommandHandler(CompletePiece)
+export class CompletePieceHandler
+  implements IInferredCommandHandler<CompletePiece> {
+  private readonly logger = new Logger(this.constructor.name);
+
+  constructor(
+    private readonly exportRepository: ExportRepository,
+    private readonly eventPublisher: EventPublisher,
+  ) {}
+
+  async execute({
+    exportId,
+    componentLocation,
+    componentId,
+  }: CompletePiece): Promise<void> {
+    const exportInstance = await this.exportRepository.find(exportId);
+
+    if (!exportInstance) {
+      this.logger.error(
+        `${CompletePieceHandler.name} could not find export ${exportId.value} to complete piece: ${componentId.value}`,
+      );
+      // TODO: could emit event as compensation action to remove the artifact
```

Comment:
in which cases could this happen? 🤔

Reply:
@hotzevzl is very unlikely - `export` could be deleted in db, or we would just have some error in the code, passing wrong exportId (i.e. passing some other ID)